ts/packages/shell/src/renderer/src/chatView.ts

```diff
@@ -718,9 +718,6 @@ export class ChatView {
         this.topDiv.className = "chat-container";
         this.messageDiv = document.createElement("div");
         this.messageDiv.className = "chat scroll_enabled";
-        this.messageDiv.addEventListener("scroll", () => {
-            this.autoScroll = false;
```

Comment:
Wouldn't that always scroll to the bottom if there are any content updates.
(That would include any dynamic updates) .

This is to turn off autoscrolling if people scroll away.  (which will reset to auto scroll when a new request started.)

Reply:
the sequence is that the event fires for the user request being added, that turns off auto scroll, then assistant message is updated but no scroll happens

we need to distinguish between when a person has scrolled and when scrolling occurs due to overflow

one way to do that might be to check whether scrollHeight === scrollTop 
don't know if that's reliable

Reply:
we resolved this by observing that without alerts there is a small window in which assistant updates can come in